packages/instant/src/redux/async_data.ts

```diff
@@ -1,22 +1,37 @@
+import * as _ from 'lodash';
+
 import { BIG_NUMBER_ZERO } from '../constants';
+import { assetUtils } from '../util/asset';
 import { coinbaseApi } from '../util/coinbase_api';
+import { errorFlasher } from '../util/error_flasher';
 
-import { ActionTypes } from './actions';
-
+import { actions } from './actions';
 import { Store } from './store';
 
 export const asyncData = {
-    fetchAndDispatchToStore: async (store: Store) => {
-        let ethUsdPrice = BIG_NUMBER_ZERO;
+    fetchEthPriceAndDispatchToStore: async (store: Store) => {
         try {
-            ethUsdPrice = await coinbaseApi.getEthUsdPrice();
+            const ethUsdPrice = await coinbaseApi.getEthUsdPrice();
+            store.dispatch(actions.updateEthUsdPrice(ethUsdPrice));
         } catch (e) {
-            // ignore
-        } finally {
-            store.dispatch({
-                type: ActionTypes.UPDATE_ETH_USD_PRICE,
-                data: ethUsdPrice,
-            });
+            const errorMessage = 'Error fetching ETH/USD price';
+            errorFlasher.flashNewErrorMessage(store.dispatch, errorMessage);
+            store.dispatch(actions.updateEthUsdPrice(BIG_NUMBER_ZERO));
+        }
+    },
+    fetchAvailableAssetDatasAndDispatchToStore: async (store: Store) => {
+        const { assetBuyer, assetMetaDataMap, network } = store.getState();
+        if (!_.isUndefined(assetBuyer)) {
+            try {
+                const assetDatas = await assetBuyer.getAvailableAssetDatasAsync();
+                const assets = assetUtils.createAssetsFromAssetDatas(assetDatas, assetMetaDataMap, network);
+                store.dispatch(actions.setAvailableAssets(assets));
+            } catch (e) {
+                const errorMessage = 'Could not find any assets';
```

Comment:
I think `tokens` may be a more familiar word to the user, but this is not a huge deal